### PR TITLE
Add Variable/Evaluate/Scope Request Handler

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

### CODE REVIEW

**Review by andxu — APPROVED** [review]

**Comment by andxu** [comment]
#65

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/StackFrameScope.java [hunk 1]
```diff
@@ -1,63 +0,0 @@
...
-*     Microsoft Corporation - initial API and implementation
-*******************************************************************************/
-
-package org.eclipse.jdt.ls.debug.adapter.variables;
-
-import java.util.Objects;
-
-import com.sun.jdi.StackFrame;
-
-public class StackFrameScope {
```
Comment (akaroml) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/StackFrameScope.java:18]: This class is changed which changes the implementation of related request handling logic. @andxu please review the new implementation.

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/ThreadObjectReference.java [hunk 2]
```diff
@@ -1,69 +0,0 @@
...
- *******************************************************************************/
-
-package org.eclipse.jdt.ls.debug.adapter.variables;
-
-import java.util.Objects;
-
-import com.sun.jdi.ObjectReference;
-import com.sun.jdi.ThreadReference;
-
-public class ThreadObjectReference {
```
Comment (akaroml) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/ThreadObjectReference.java:19]: This class is changed which changes the implementation of related request handling logic. @andxu please review the new implementation.

**Review by akaroml — CHANGES_REQUESTED** [review]

**Review by akaroml — APPROVED** [review]

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/StackFrameScope.java (hunk 1 above)
Comment (testforstephen) [reply org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/StackFrameScope.java:18]: StackFrameScope and ThreadObjectReference are similar, they could be merged into a neutral VariableProxy. @andxu has agreed that.

**Pull request merged**